### PR TITLE
[MAINTENANCE] Include defaults when converting `Expectation` to `ExpectationConfiguration`

### PR DESCRIPTION
**Pull request by cdkini**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=9370&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=9370&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: great_expectations/expectations/expectation.py [hunk 1]
```diff
@@ -1750,7 +1758,7 @@ class ColumnMapExpectation(BatchExpectation, ABC):
     map_metric: ClassVar[Optional[str]] = None
     domain_keys: ClassVar[Tuple[str, ...]] = (
         "batch_id",
-        "table",
+        "table",  # Why is this here???
```
Comment (cdkini) [inline great_expectations/expectations/expectation.py:1761]: Seems weird that we put table in a column-specific Expectation no?

File: great_expectations/expectations/expectation.py [hunk 2]
```diff
@@ -1226,6 +1226,14 @@ def validate_(  # noqa: PLR0913
     @property
     def configuration(self) -> ExpectationConfiguration:
         kwargs = self.dict(exclude_defaults=True)
+
+        # Always add back domain-specific keys if there's a value associated with them (even if they are defaults)
+        for key in self.domain_keys:
+            if key not in kwargs and key in self.__fields__:
+                default = self.__fields__[key].default
+                if default:
+                    kwargs[key] = default
```
Comment (cdkini) [inline great_expectations/expectations/expectation.py:1235]: Not my favorite piece of logic but if I have this custom Expectation:
```python
 class ExpectColumnAgesToBeLegalAdult(gxe.ExpectColumnValuesToBeBetween):
        column: str = "ages"
        min_value: int = 18
        description: str = "column values must be a legal adult age"
```

the current `configuration` property will remove all defaults, resulting in empty kwargs. Without kwargs in the config, renderers are unable to function properly. They need to know what column to associate the object with at a minimum so I've opted to use this logic to ensure we always have it when we need it.

Note that this issue is a very small edge case concerning simple custom expectations that override domains. If we want to encourage users to always provide a domain or we don't like the logic I've used here, let's forgo my changes

File: tests/render/test_column_section_renderer.py [hunk 3]
```diff
@@ -85,7 +85,18 @@ class ExpectColumnAgesToBeLegalAdult(gxe.ExpectColumnValuesToBeBetween):
         min_value: int = 18
         description: str = "column values must be a legal adult age"
 
-    return ExpectColumnAgesToBeLegalAdult()
+    expectation = ExpectColumnAgesToBeLegalAdult()
+    try:
+        kwargs = expectation.configuration.kwargs
+        assert kwargs.get("column") == "ages"
+        assert kwargs.get("min_value") is None
+        assert kwargs.get("description") is None
```
Comment (cdkini) [inline tests/render/test_column_section_renderer.py:93]: With the above logic, we should see `column` put back into `configuration.kwargs` but the other two overriden params left alone

**Review by tyler-hoffman — COMMENTED** [review]
I agree that putting this logic in our base Expectation class isn't ideal. Unsure if the tradeoff is worth is, so I'm going to avoid approving or requesting changes until we chat about it more. I'm curious if it's worth exploring some way for authors to indicate to us that they want to opt into this behavior, maybe via a flag in pydantic Fields or something?

**Comment by cdkini** [comment]
Closing due to the conclusion that the effort/complexity isn't worth it for a niche use-case. We should discourage users from overriding domain keys
